src/H5Oint.c

```diff
@@ -2684,15 +2694,17 @@ H5O__visit(H5G_loc_t *loc, const char *obj_name, H5_index_t idx_type, H5_iter_or
         HGOTO_ERROR(H5E_ID, H5E_CANTREGISTER, FAIL, "unable to register visited object");
 
     /* Make callback for starting object */
-    if ((ret_value = op(obj_id, ".", &oinfo, op_data)) < 0)
+    /* if ((ret_value = op(obj_id, ".", oinfop, op_data)) < 0)
+     */
+    if ((ret_value = op(obj_id, ".", oinfop, op_data)) < 0)
```

Comment:
This won't work when the application's fields don't include BASIC.  The oinfop will be pointing at the wrong struct.

You can fix it by moving the block of code with the query on the internal oinfo below this point.  (I suggest below the check for H5_ITER_CONT)

Reply:
Ah, actually, it was oinfo there before I switched to oinfop, but now I see that oinfo would be the correct one.  I'll switch it back unless you disagree.

Reply:
That would also work :-)

Reply:
I'll note that no tests were failing when this was wrong.   You should probably add a test that fails (now), so you can be certain it's working after you correct it.

Reply:
Hmm, I did add one test that used H5O_INFO_META_SIZE, so no BASIC...